### PR TITLE
feat: initial repo setup

### PR DESCRIPTION
**Pull request by jpower432**
## Objectives

Add an initial structure as a jumping off point

## Important Changes added here

- Initialized Go module
- Adds initial code structure under `cmd` complytime and plugin
- Add LICENSE and gitignore
- Add initial Makefile with some general helpful commands for testing, building, and formatting with Go tools

## Dependencies added

[Cobra](https://github.com/spf13/cobra)
[Testify](https://github.com/stretchr/testify)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*.swp
+*.swo
+*~
+.vscode
+
+# Output of gorelease
+dist
+
+# ignore output by test
+/**/_test
```
Comment (marcusburghardt) [inline .gitignore]: Could you include an final empty line, please? Maybe it would be good create an initial style guide doc.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+format:
+	go fmt ./...
+.PHONY: format
+
+vet:
+	go vet ./...
+.PHONY: vet
+
+all: clean vendor test-unit build
+.PHONY: all
```
Comment (marcusburghardt) [inline Makefile]: Missing empty line.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
+GO_BUILD_PACKAGES := ./cmd/...
+GO_BUILD_BINDIR :=./bin
+GIT_COMMIT := $(or $(SOURCE_GIT_COMMIT),$(shell git rev-parse --short HEAD))
+GIT_TAG :="$(shell git tag | sort -V | tail -1)"
+
+GO_LD_EXTRAFLAGS :=-X github.com/complytime/complytime/version.version="$(shell git tag | sort -V | tail -1)" \
```
Comment (marcusburghardt) [inline Makefile]: Can we use `GIT_TAG` defined above here?

**Review by marcusburghardt — COMMENTED** [review]
I have only some minor comments.

File: .gitignore (hunk 1 above)
Comment (jpower432) [reply .gitignore]: @marcusburghardt What do you think about adding a pre-commit hook to check for this? I don't disagree that we should have a style guide, but automating will reduce errors and burden on the reviewer.

File: Makefile (hunk 3 above)
Comment (jpower432) [reply Makefile]: Good catch! I will fix this.

File: .gitignore (hunk 1 above)
Comment (marcusburghardt) [reply .gitignore]: I think this would be great. We can keep an style guide for reference but automating the checks would indeed save us time.

**Review by marcusburghardt — APPROVED** [review]
LGTM. I have tested locally. Thanks @jpower432

**Review by huiwangredhat — APPROVED** [review]
LGTM. I tested locally about the Makefile, ignoring bin, complytime cmd, and version_test. Except the `test` is failed due to my local system, all of them worked well.

File: cmd/complytime/cli/root.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package cli
+
+import (
```
Comment (qduanmu) [inline cmd/complytime/cli/root.go:3]: There will be a specific folder with the suffix ‘-plugin’ under ‘cmd’ folder for each plugin, and these plugins will be imported here and run as a sub-command of complytime, right?

**Comment by vojtapolasek** [comment]
It looks good, as long as my beginner Golang skills are taken into account. I use it as learning material :-)

File: cmd/complytime/cli/root.go (hunk 4 above)
Comment (jpower432) [reply cmd/complytime/cli/root.go:3]: @qduanmu The plugin and ComplyTime will be compiled into separate binaries. We are co-locating them in this repository at the moment while the project is under early development.

**Review by qduanmu — APPROVED** [review]
LGTM

**Comment by jpower432** [comment]
> LGTM. I tested locally about the Makefile, ignoring bin, complytime cmd, and version_test. Except the `test` is failed due to my local system, all of them worked well.

@huiwangredhat I actually think that is an issue with the test. Might be better to remove system specific information there like Go version or Platform. It might need a little rework.

**Comment by jpower432** [comment]
@huiwangredhat Is this PR blocked by #3 waiting for OpenSSF scanning results?

**Review by gvauter — APPROVED** [review]
lgtm

**Comment by huiwangredhat** [comment]
> @huiwangredhat Is this PR blocked by #3 waiting for OpenSSF scanning results?

I set the requirements in CPLYTM-293, which include scorecard scanning. I'd like to remove this requirement. The scorecard scan workflow will be added via the PR3. You could merge it to the main branch first.

**Review by d10n — APPROVED** [review]

**Pull request merged**